tools/src/edm4hep2json.cxx

```diff
@@ -120,20 +119,8 @@ int main(int argc, char** argv) {
     outFilePath = std::filesystem::path(outFileStr + ".edm4hep.json");
   }
 
-  bool legacyReader = false;
-  {
-    std::unique_ptr<TFile> inFile(TFile::Open(inFilePath.c_str(), "READ"));
-    legacyReader = !inFile->GetListOfKeys()->FindObject("podio_metadata");
-  }
-
-  if (legacyReader) {
-    std::cout << "WARNING: Reading legacy file, some collections might not be recognized!" << std::endl;
-    return read_frames<podio::ROOTLegacyReader>(inFilePath, outFilePath, requestedCollections, requestedEvents,
-                                                frameName, nEventsMax, verboser);
-  } else {
-    return read_frames<podio::ROOTReader>(inFilePath, outFilePath, requestedCollections, requestedEvents, frameName,
-                                          nEventsMax, verboser);
-  }
+  return read_frames<podio::ROOTReader>(inFilePath, outFilePath, requestedCollections, requestedEvents, frameName,
+                                        nEventsMax, verboser);
```

Comment:
Maybe switch this directly to the `podio::Reader`?

Reply:
To do this one option to change the category of the dumped frame has to be changed, but maybe that option is never used. Would that be fine @kjvbrt?

Reply:
Why doe we need to change that? The `Reader` also has a `readFrame` method [doc](https://key4hep.web.cern.ch/podio/cpp_api/class/classpodio_1_1_reader.html#_CPPv4N5podio6Reader9readFrameERKNSt6stringE6size_t). We could get rid of a few double Frame constructions, since the `Reader` already returns a Frame.

Reply:
Yes, fixed now